src/scripts/npc/npc.gd

```diff
@@ -25,53 +25,46 @@ func _ready():
+	toggle_enter_button()
 	boss_fight_dialog()
 
 func _input(_event):
-	if (self.has_node("EnterButton")):
-		$EnterButton.visible = active
+	toggle_enter_button()
 	npc_dialog()
 
 func npc_dialog():
 	if (Globle.player_character == "Angela" and get_node_or_null('DialogNode') == null and Input.is_action_just_pressed("ui_accept") and active):
-		if get_node_or_null('DialogNode') == null:
-			if Input.is_action_just_pressed("ui_accept") and active: 
-				match(character_name):
-					"Mia":
-						match(mia_dialog_value):
-							(0):
-								Globle.player_active = false
-								commence_dialog('Mia_Angela_timeline')
-								mia_dialog_value = process_dialog_value(npc_angela_rivet_dialog_value,2)
-							(2):
-								mia_dialog_value = process_dialog_value(npc_angela_rivet_dialog_value, 0)
-								Globle.player_active = true
-					"NPC_Angela_Rivet":
-						match(npc_angela_rivet_dialog_value):
-							(0):
-								commence_dialog('timeline_Rivet_npc_1')
-								npc_angela_rivet_dialog_value = process_dialog_value(npc_angela_rivet_dialog_value, 2)
-					"Shark_man":
-						match(shark_man_dialog_value):
-							(0):
-								Globle.arena_menu_open = false
-								Globle.player_active = false
-								commence_dialog('timeline_Angela_shark_man')
-								shark_man_dialog_value = process_dialog_value(npc_angela_rivet_dialog_value, 2)
-							(1):
-								shark_man_dialog_value = process_dialog_value(npc_angela_rivet_dialog_value, 0)
-								Globle.player_active = true
-								Globle.arena_menu_open = true
-					"F44":
-						print("Ce soir la vie n'est plus un enfer pour 1 fois.")
-	if (Globle.player_character == "Rivet" and get_node_or_null('DialogNode') == null and Input.is_action_just_pressed("ui_accept") and active == true):
 		match(character_name):
 			"Mia":
 				match(mia_dialog_value):
 					(0):
-						Globle.player_active = false
+						commence_dialog('Mia_Angela_timeline')
+						mia_dialog_value = process_dialog_value(npc_angela_rivet_dialog_value,2)
+					(2):
+						mia_dialog_value = process_dialog_value(npc_angela_rivet_dialog_value, 0)
+						Globle.player_active = true
+			"NPC_Angela_Rivet":
+				match(npc_angela_rivet_dialog_value):
+					(0):
+						commence_dialog('timeline_Rivet_npc_1')
+						npc_angela_rivet_dialog_value = process_dialog_value(npc_angela_rivet_dialog_value, 2)
+			"Shark_man":
+				match(shark_man_dialog_value):
+					(0):
+						Globle.arena_menu_open = false
+						commence_dialog('timeline_Angela_shark_man')
+						shark_man_dialog_value = process_dialog_value(npc_angela_rivet_dialog_value, 2)
+					(1):
+						shark_man_dialog_value = process_dialog_value(npc_angela_rivet_dialog_value, 0)
+						Globle.player_active = true
+						Globle.arena_menu_open = true
+			"F44":
+				print("Ce soir la vie n'est plus un enfer pour 1 fois.")
+	if (Globle.player_character == "Rivet" and get_node_or_null('DialogNode') == null and Input.is_action_just_pressed("ui_accept") and active):
```

Comment:
when trying to go to angela as rivet it is not activating for some reason.
![image](https://github.com/user-attachments/assets/728e603a-ef4d-4823-84ff-ce03110037e7)